internal/pkg/providers/azure/azure.go

```diff
@@ -0,0 +1,238 @@
+package azure
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"strconv"
+	"strings"
+
+	"github.com/Azure/azure-sdk-for-go/profiles/latest/network/mgmt/network"
+	"github.com/Azure/azure-sdk-for-go/services/resources/mgmt/2019-05-01/resources"
+
+	"github.com/Azure/go-autorest/autorest/azure/auth"
+	"github.com/Azure/go-autorest/autorest/to"
+	"github.com/mitchellh/mapstructure"
+	"github.com/sirupsen/logrus"
+	"github.com/stakater/Whitelister/internal/pkg/config"
+	clientset "k8s.io/client-go/kubernetes"
+
+	"github.com/stakater/Whitelister/internal/pkg/utils"
+)
+
+// Azure provider class implementing the Provider interface
+type Azure struct {
+	ClientSet                 clientset.Interface
+	SubscriptionID            string
+	ClientID                  string
+	ClientSecret              string
+	TenantID                  string
+	ResourceGroupName         string
+	RemoveRule                bool
+	KeepRuleDescriptionPrefix string
+}
+
+// GetName Returns name of provider
+func (a *Azure) GetName() string {
+	return "Azure"
+}
+
+// Init initializes the Azure Provider Configuration like client id and client secret
+func (a *Azure) Init(params map[interface{}]interface{}, clientSet clientset.Interface) error {
+	a.ClientSet = clientSet
+	err := mapstructure.Decode(params, &a) //Converts the params to Azure struct fields
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+// WhiteListIps - Get List of IP addresses to whitelist
+func (a *Azure) WhiteListIps(filter config.Filter, ipPermissions []utils.IpPermission) error {
+
+	resourceGroups, err := fetchResourceGroups(a, filter)
```

Comment:
Shouldn't we fetch the resource group based on `resourceGroupName` that the user provides?